compiler/rustc_passes/src/stability.rs

```diff
@@ -816,48 +816,20 @@ impl<'tcx> Visitor<'tcx> for Checker<'tcx> {
     }
 }
 
-/// Check whether the path is a re-export of an unstable item, that has also been marked as unstable under the same feature name.
-fn is_unstable_reexport<'a, 'tcx>(
+/// Check whether a path is a `use` item
+fn is_path_reexport<'tcx>(
     tcx: TyCtxt<'tcx>,
-    path: &'a hir::Path<'tcx>,
     id: hir::HirId,
 ) -> bool {
     // Get the LocalDefId so we can lookup the item to check the kind.
     let Some(def_id) = tcx.hir().opt_local_def_id(id) else { return false};
 
-    let Some(stab) = tcx.stability().local_stability(def_id) else {
-        return false;
-    };
-
     // If this is a path that isn't a use, we don't need to do anything special
     if !matches!(tcx.hir().item(hir::ItemId { def_id }).kind, ItemKind::Use(..)) {
         return false;
     }
 
-     let Some(def_id) = path.res.opt_def_id() else {
-        // If the path doesn't resolve to anything, there's nothing more to check.
-        return false;
-    };
-
-    let method_span = path.segments.last().map(|s| s.ident.span);
-
-    let decision = tcx.eval_stability(def_id, Some(id), path.span, method_span);
-
-    // If we are going to deny the usage, check whether it's a re-export first with the same unstable feature.
-    if let rustc_middle::middle::stability::EvalResult::Deny { feature, .. } = decision {
-        if stab.level.is_unstable() && feature == stab.feature {
```

Comment:
This check was necessary, it ensured that the `use` item itself had an `#[unstable]` attribute.
Without it any third-party crate can reexport something unstable from the standard library on stable compiler.